### ISSUE TITLE
feat(experiments): SAM3 Vanilla + Hybrid + VesselFM GPU runs for R3b/R3c manuscript results

### ISSUE THREAD

**Issue by petteriTeikari** [petteriTeikari/minivess-mlops#734]
## Summary

SAM3 Vanilla, SAM3 Hybrid, and VesselFM (external data only — DeepVess/TubeNet, NOT MiniVess) GPU runs are required to complete manuscript results R3b (multi-model DSC/clDice comparison) and R3c (VesselFM external validation). Without R3b the paper cannot be submitted. Target: RTX 4090 on RunPod, ~$15 total.

### Infrastructure: FULLY READY
All adapters, configs, SkyPilot YAMLs, and Docker images are implemented and tested.

**Adapters (all implemented, unit tested):**
- `src/minivess/adapters/sam3_vanilla.py` — V1: frozen ViT-32L encoder, lightweight Conv decoder (66K trainable params)
- `src/minivess/adapters/sam3_hybrid.py` — V3: frozen ViT-32L + gated DynUNet-3D fusion
- `src/minivess/adapters/sam3_topolora.py` — TopoLoRA variant (topology-aware LoRA)
- `src/minivess/adapters/vesselfm.py` — VesselFM adapter (external evaluation only)
- `src/minivess/adapters/sam3_backbone.py` — Shared SAM3 ViT-32L + FPN neck; SDPA + BF16
- `src/minivess/adapters/sam3_decoder.py` — SAM3 decoder module
- `src/minivess/adapters/sam3_feature_cache.py` — Feature caching for inference efficiency
- `src/minivess/adapters/sam3_vram_check.py` — Pre-flight VRAM enforcement (>=16 GB for SAM3)
- `src/minivess/adapters/model_builder.py` — `build_adapter()` factory

**Experiment Configs (11 YAML files):**
- `configs/experiment/sam3_vanilla_baseline.yaml`, `sam3_vanilla_debug.yaml`, `smoke_sam3_vanilla.yaml`
- `configs/experiment/sam3_hybrid_fusion.yaml`, `sam3_hybrid_debug.yaml`, `sam3_hybrid_full.yaml`, `smoke_sam3_hybrid.yaml`, `smoke_sam3_hybrid_cloud.yaml`
- `configs/experiment/sam3_topolora_topology.yaml`, `sam3_topolora_debug.yaml`, `smoke_sam3_topolora.yaml`
- `configs/experiment/vesselfm_zeroshot.yaml`, `vesselfm_finetune.yaml`, `smoke_vesselfm.yaml`

**Model Profiles:**
- `configs/model/sam3_vanilla.yaml`, `sam3_hybrid.yaml`, `vesselfm.yaml`
- `configs/model_profiles/sam3_vanilla.yaml`, `sam3_hybrid.yaml`, `vesselfm.yaml`

**SkyPilot Deployment (10 YAML files):**
- `deployment/skypilot/smoke_test_gpu.yaml` — RunPod RTX 4090
- `deployment/skypilot/smoke_test_gcp.yaml` — GCP L4 spot
- `deployment/skypilot/train_factorial.yaml` — Full factorial runs
- `deployment/skypilot/train_production.yaml` — Production runs
- `deployment/skypilot/train_hpo.yaml`, `hpo_grid_worker.yaml`
- `deployment/skypilot/dev_runpod.yaml`, volume YAMLs, mamba smoke test

**DynUNet baseline:** COMPLETE (3 folds, all losses — see `knowledge-graph/experiments/dynunet_loss_variation_v2.yaml`)

### Runs: BLOCKED — Awaiting GPU Budget Allocation
- **SAM3 Vanilla**: PENDING GPU run (adapter + configs verified locally)
- **SAM3 Hybrid**: PENDING GPU run (adapter + configs verified locally)
- **SAM3 TopoLoRA**: PENDING GPU run (adapter + configs verified locally)
- **VesselFM**: PENDING GPU run (external-only evaluation — data leakage constraint)

## What's Missing

1. **Cloud GPU budget allocation** (~$15 RunPod or ~$18 GCP L4 spot)
2. **Actual GPU execution** of SAM3 Vanilla, Hybrid, TopoLoRA, and VesselFM
3. **Results population** in `knowledge-graph/experiments/current_best.yaml`
4. **Manuscript numbers** in `docs/manuscript/latent-methods-results/results/results-03-models.tex`
5. **R3b/R3c status update** in `knowledge-graph/manuscript/results.yaml`

### Core Adapters
- `src/minivess/adapters/sam3_vanilla.py` — Frozen ViT-32L encoder + Conv decoder
- `src/minivess/adapters/sam3_hybrid.py` — Frozen ViT-32L + gated DynUNet-3D fusion
- `src/minivess/adapters/sam3_topolora.py` — Topology-aware LoRA variant
- `src/minivess/adapters/vesselfm.py` — External evaluation only (data leakage)
- `src/minivess/adapters/sam3_backbone.py` — Shared ViT-32L + FPN, SDPA + BF16
- `src/minivess/adapters/sam3_vram_check.py` — VRAM >=16 GB enforcement
- `src/minivess/adapters/model_builder.py` — Factory dispatch
- `src/minivess/adapters/CLAUDE.md` — VRAM tables, dimension order, BF16 requirements

### SkyPilot & Cloud
- `deployment/skypilot/smoke_test_gpu.yaml` — RunPod RTX 4090 target
- `deployment/skypilot/smoke_test_gcp.yaml` — GCP L4 spot target
- `configs/cloud/yaml_contract.yaml` — Golden reference: L4 (GCP), RTX4090/5090/3090 (RunPod)
- `configs/cloud/gcp_spot.yaml` — GCP L4 spot config, europe-north1

### Launch Commands
```bash
# RunPod (recommended — cheapest, instant provisioning)
make smoke-test-gpu MODEL=sam3_vanilla
make smoke-test-gpu MODEL=sam3_hybrid
make smoke-test-gpu MODEL=vesselfm

# Or directly via SkyPilot:
sky jobs launch deployment/skypilot/smoke_test_gpu.yaml \
  --env-file .env --env MODEL_FAMILY=sam3_vanilla -y
```

## Blockers

| Blocker | Status | Notes |
|---------|--------|-------|
| Cloud GPU budget (~$15-18) | **BLOCKED** | Primary blocker — awaiting allocation |
| SAM3 adapters implemented | DONE | All 3 variants + VesselFM |
| SkyPilot YAML configs | DONE | RunPod + GCP targets |
| Docker image pushed | DONE | `petteriteikari/minivess-base:latest` on Docker Hub |
| RunPod Network Volume | DONE | `minivess-dev` with MiniVess data |
| #789 (factorial experiment) | DOWNSTREAM | This issue feeds results into factorial design |
| #751 (Docker pull optimization) | NICE-TO-HAVE | Reduces GCP smoke test wall time |

## Priority Context

**CRITICAL PATH for Nature Protocols paper.** R3b (multi-model DSC/clDice comparison) is a hard blocker for manuscript submission. Without SAM3 and VesselFM results, the paper cannot demonstrate the platform's multi-model capability — the central contribution.

**T4 BANNED** (Turing, no BF16) — use RTX 4090 (Ada Lovelace, BF16). SAM3 requires BF16 — FP16 overflows during validation (MONAI #4243). Estimated cost: ~$5-8 for SAM3 runs, ~$3-5 for VesselFM. AMP validation must be OFF for SAM3.

**VesselFM data leakage constraint:** VesselFM was pre-trained on MiniVess — evaluating on MiniVess is data leakage. Must use DeepVess (Cornell eCommons) or TubeNet 2PM (UCL Figshare).

## Planning References

| Document | Relevance |
|----------|-----------|
| `docs/planning/kg-manuscript-execution-plan.yaml` | Master execution plan |
| `docs/planning/sam3-training-reference.md` | SAM3 training hyperparameters |
| `docs/planning/sam3-val-loss-final-report.md` | SAM3 validation loss debugging |
| `docs/planning/sam3-nan-loss-fix.md` | NaN loss root cause (FP16 overflow) |
| `docs/planning/factorial-design-demo-experiment-plan.md` | Factorial experiment design |

## Acceptance Criteria

- [ ] SAM3 Vanilla: full run (3 folds x 100 epochs), metrics saved to mlruns
- [ ] SAM3 Hybrid: full run (3 folds x 100 epochs), metrics saved to mlruns
- [ ] SAM3 TopoLoRA: full run (3 folds x 100 epochs), metrics saved to mlruns
- [ ] VesselFM: evaluate on DeepVess + TubeNet ONLY (NOT MiniVess — data leakage)
- [ ] `knowledge-graph/experiments/current_best.yaml` updated with results
- [ ] `knowledge-graph/manuscript/results.yaml` R3b, R3c updated from BLOCKED to complete
- [ ] `docs/manuscript/latent-methods-results/results/results-03-models.tex` updated with real numbers

## References

- [Wittmann et al. (2024). "VesselFM: A Foundation Model for Universal 3D Blood Vessel Segmentation." *arXiv:2411.17386*](https://arxiv.org/abs/2411.17386)
